third_party/libusb/webport/src/libusb-to-webusb-adaptor.js

```diff
@@ -292,6 +306,170 @@ function getLibusbJsEndpointType(webusbEndpointType) {
+  // pointers to the LibusbJs objects that correspond to the currently parsed
+  // interface/endpoint descriptor.
+  /** @type {!LibusbJsInterfaceDescriptor|undefined} */
+  let currentInterface = undefined;
+  /** @type {!LibusbJsEndpointDescriptor|undefined} */
+  let currentEndpoint = undefined;
+  for (let offset = 0; offset < totalLength;) {
+    const descriptorLength = descriptors.getUint8(offset);
+    if (descriptorLength < 2) {
+      // Ignore the invalid (too short) descriptor.
+      offset += descriptorLength;
+      continue;
+    }
+    if (offset + descriptorLength > totalLength) {
+      // Ignore the truncated descriptor.
+      break;
+    }
+    const descriptorType = descriptors.getUint8(offset + 1);
+    switch (descriptorType) {
+      case CONFIGURATION_DESCRIPTOR_TYPE: {
+        if (descriptorLength < CONFIGURATION_DESCRIPTOR_LENGTH) {
+          // Ignore the invalid (too short) descriptor.
+          break;
+        }
+        // The current item is the configuration descriptor. There should be
+        // only one such item returned and it should refer to
+        // `libusbJsConfiguration`, so nothing needs to be done here.
+        break;
+      }
+      case INTERFACE_DESCRIPTOR_TYPE: {
+        if (descriptorLength < INTERFACE_DESCRIPTOR_LENGTH) {
+          // Ignore the invalid (too short) descriptor.
+          break;
+        }
+        // The current item is the interface descriptor. Parse the interface
+        // number and switch the pointer to the corresponding
+        // `LibusbJsInterfaceDescriptor`.
+        const interfaceNumber = descriptors.getUint8(offset + 2);
+        currentInterface = libusbJsConfiguration['interfaces'].find(
+            libusbJsInterface =>
+                libusbJsInterface['interfaceNumber'] === interfaceNumber);
+        currentEndpoint = undefined;
+        break;
+      }
+      case ENDPOINT_DESCRIPTOR_TYPE: {
+        if (descriptorLength < ENDPOINT_DESCRIPTOR_LENGTH) {
+          // Ignore the invalid (too short) descriptor.
+          break;
+        }
+        // The current item is the endpoint descriptor. Parse the endpoint
+        // address and switch the pointer to the corresponding
+        // `LibusbJsEndpointDescriptor`.
+        const endpointAddress = descriptors.getUint8(offset + 2);
+        if (currentInterface) {
+          currentEndpoint = currentInterface['endpoints'].find(
+              libusbJsEndpoint =>
+                  libusbJsEndpoint['endpointAddress'] === endpointAddress);
+        } else {
+          currentEndpoint = undefined;
+        }
+        break;
+      }
+      default: {
+        // The current item is an unknown descriptor, so add it as extraData to
+        // the current (the most nested one) LibusbJs object.
+        const targetObject = currentEndpoint ?
+            currentEndpoint :
+            (currentInterface ? currentInterface : libusbJsConfiguration);
```

Comment:
I think JavaScript will let you write this as,

```suggestion
        const targetObject = currentEndpoint || currentInterface || libusbJsConfiguration;
```

Reply:
Done, thank you.